include/mega/command.h

```diff
@@ -1205,6 +1205,44 @@ class MEGA_API CommandFolderLinkInfo: public Command
     CommandFolderLinkInfo(MegaClient*, handle);
 };
 
+class MEGA_API CommandSyncPut : public Command
+{
+public:
+    void procresult();
+
+    // Register a new Sync
+    CommandSyncPut(MegaClient* client, SyncType type, handle nodeHandle, const std::string& localFolder, const std::string& deviceId, const std::string& syncName, int state, int subState, const std::string& extraData);
+
+    // Update a Sync
+    // Params that keep the same value are passed with invalid value to avoid to send to the server
+    // Invalid values:
+    // - type: SyncType::INVALID
+    // - nodeHandle: UNDEF
+    // - localFolder: nullptr
+    // - deviceId: nullptr
+    // - SyncName: nullptr
+    // - state: -1
+    // - subState: -1
+    // - extraData: nullptr
+    CommandSyncPut(MegaClient* client, handle syncId, SyncType type, handle nodeHandle, const char* localFolder, const char* deviceId, const char* syncName, int state, int subState, const char* extraData);
```

Comment:
Thinking-out-loud: Perhaps it'll be worth it to separate the 2 use cases: register vs update in 2 different commands, although the both entail "sp".
We might also like to have a separated App callback: `syncupdate_result`.
Later on, we might need MegaApiImpl to fire diffferent callbacks for `syncput_result` vs `syncupdate_result`, in case apps require to manage/show heartbeated syncs

Reply:
I'm not sure how it's going to be implemented at app side, if you consider that have two commands be worth. I'll implement it.
I was thinking that many of this stuff is going to keep internally

Reply:
I kind of agree with both of you. Nevertheless, having two separate callbacks for registering and for updating could be more flexible in the future, so I think we should go that way.